### ISSUE TITLE
Investigate native Websocket support

### ISSUE THREAD

**Issue by bkimminich** [juice-shop/juice-shop#339]
Express and Angular should both be able to handle Websockets natively(?) making the dependency on `socket.io` unnecessary.

This should be investigated as it would remove a rather heavyweight dependency.

**Comment by supra08** [comment]
I believe this is a nice idea. Express does support Websockets natively, removing dependancy of `socket.io`. It can be achieved using the `ws` module of Express. For the sake of comparison between Express and socket.io, have a look at this [link](https://stackoverflow.com/a/38558531/2590859).
This [article](https://medium.com/factory-mind/websocket-node-js-express-step-by-step-using-typescript-725114ad5fe4) provides a sample implementation for the same.
For Angular, I did not find any particular library, except for [one](https://github.com/PatrickJS/angular-websocket). But I could not find a genuine comparison with the `socket.io client` for this, so can't debate on it performance wise. I found this [article](https://www.smartspate.com/websockets-in-angular/) helpful too. @bkimminich, @J12934 ?
